pkg/agent/baker.go

```diff
@@ -704,3 +705,25 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		},
 	}
 }
+
+// ParseVMSSCSEMessage parses the raw VMSS CSE output
+func ParseVMSSCSEMessage(message string) (datamodel.VMSSInstanceViewCSEStatus, error) {
+	var cseStatus datamodel.VMSSInstanceViewCSEStatus
+	var rerr error
+	start := strings.Index(message, "[stdout]") + len("[stdout]")
+	end := strings.Index(message, "[stderr]")
+	if end > start {
+		rawInstanceViewInfo := message[start:end]
+		err := json.Unmarshal([]byte(rawInstanceViewInfo), &cseStatus)
+		if err != nil || cseStatus.ExitCode == "" {
+			// RP needs regex "vmssInstanceErrorCode=" to parse the error, please inform RP if you would like to change the regex.
```

Comment:
I suggest we do not assume there is only one user of this library, can we say  the error text is part of contract, and client is relying on it?

Reply:
fix it